system/jlib/jmutex.hpp

```diff
@@ -165,23 +184,30 @@ class jlib_decl NamedMutex
     bool lockWait(unsigned timeout);
     void unlock();
 private:
-    Mutex threadmutex;
+    TimedMutex threadmutex;
     char *mutexfname;
 };
 
+template<class T> class jlib_decl MutexBlock
+{
+private:
+    T &mutex;
+public:
+    MutexBlock<T>(T &m) : mutex(m) { mutex.lock(); };
```

Comment:
I think the <T> is not needed/correct on the constructor, and is causing compile errors for wasmembed